### ISSUE TITLE
"The AzureAD Authentication is not correctly configured." warning when Azure AD is configured from the environment

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#14050]
### Describe the bug

You get a "The AzureAD Authentication is not correctly configured." warning logged, coming from [here](https://github.com/OrchardCMS/OrchardCore/blob/9c22db76f5f6299d5d2abf69fde6aa19acc64e3b/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADSettingsConfiguration.cs#L52), when Azure AD settings are configured from the environment, not site settings, via [`ConfigureAzureADSettings()`](https://github.com/OrchardCMS/OrchardCore/blob/9c22db76f5f6299d5d2abf69fde6aa19acc64e3b/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Extensions/OrchardCoreBuilderExtensions.cs#L21).

### To Reproduce
1. Configure Azure AD settings with `ConfigureAzureADSettings()`.
2. Launch the app.
3. Observe the warning being logged immediately.

### Expected behavior
No warning is logged when the configuration comes from the environment. However, a warning should be logged if no available configuration is correct.

**Comment by ns8482e** [comment]
Yes! warning message is misleading, however the order is correct. Settings are loaded first from settings by Configure and then overridden from appsettings config via `PostConfigure`  

So error should state what it finds i.e. `The AzureAD Authentication is not configured in site settings.`

**Comment by Piedone** [comment]
If the feature is configured correctly, regardless of where the config is coming from, then there should be no warning at all.

**Comment by hishamco** [comment]
Is this fixed in the related PR, or shall I submit a PR?

**Comment by Piedone** [comment]
It's isn't, and we won't do https://github.com/OrchardCMS/OrchardCore/issues/15310.
